src/main/java/com/hida/ArkMinter.java

```diff
@@ -0,0 +1,251 @@
+            }
+            String temp = buffer.toString();
+            this.ID_ARRAY.add(NAAN + this.PREFIX + temp);
+        }
+        return convertListToJson();
+    }
+
+    /**
+     * genIDcustom() Method description: This generates an customized ID format.
+     * User would give a string that excepts 6 letters: {d, l, u, m, e, a}.
+     * Similar to the Mode settings in genIDauto() method. d:
+     *
+     * <p>
+     * DIGIT: {0123456789} Digit values only.</p>
+     * <p>
+     * LOWERCASE: {abc...xyz} Lowercase letters only. </p>
+     * <p>
+     * UPPERCASE: {ABC...XYZ} Uppercase letters only.</p>
+     * <p>
+     * MIXCASE: {abc...XYZ} Lowercase and Uppercase letters only. </p>
+     * <p>
+     * EXTENDED: {012...xyz} Digit values and Lowercase letters only. </p>
+     * <p>
+     * ALPHANUMERIC: {012...abc...XYZ} Digit values, Lowercase and Uppercase
+     * letters.</p>
+     *
+     * @param isRandom - generates ids based on whether or not user requested
+     * random strings
+     * @return - A reference to a JsonObject that contains Json list of ids
+     */
+    @Override
+    public JsonObject genIdCustom(boolean isRandom) {
+        String format = this.FORMAT.toLowerCase();
+
+        double index = 0.0D;
+        for (int j = 0; j < this.AMOUNT; j++) {
+            StringBuilder buffer = new StringBuilder();
+            for (int i = 0; i < format.length(); i++) {
+                char c = format.charAt(i);
+                if (isRandom) {
+                    switch (c) {
+                        case 'd':
+                            index = Math.random() * 10.0D;
+                            buffer.append(((String) this.tokenMaps.get("DIGIT")).charAt((int) index));
+                            break;
+                        case 'l':
+                            index = Math.random() * 26.0D;
+                            buffer.append(((String) this.tokenMaps.get("LOWERCASE")).charAt((int) index));
+                            break;
+                        case 'u':
+                            index = Math.random() * 26.0D;
+                            buffer.append(((String) this.tokenMaps.get("UPPERCASE")).charAt((int) index));
+                            break;
+                        case 'm':
+                            index = Math.random() * 52.0D;
+                            buffer.append(((String) this.tokenMaps.get("MIXCASE")).charAt((int) index));
+                            break;
+                        case 'e':
+                            index = Math.random() * 36.0D;
+                            buffer.append(((String) this.tokenMaps.get("EXTENDED")).charAt((int) index));
+                            break;
+                        case 'a':
+                            index = Math.random() * 62.0D;
+                            buffer.append(((String) this.tokenMaps.get("ALPHANUMERIC")).charAt((int) index));
+                    }
+                } else {

```

Comment:
when `isRandom` is false, I don't think this will really work.  Think what if format is `dddddddddd` (10 `d`s).  Then each ID generated is exactly the the same...